### PR TITLE
ch569w-evt: add usbhs device mode driver

### PR DESCRIPTION
**Pull request by emuzit**
* usbd driver tested with cdc_vcom, internal loopback
  (can't run both MSH & usbd due to 16KB RAM limitation)
* reduce usrstack & main thread stack size for usbd test
* ch56x_uart.c : iron out UART0_PIN_ALT assignment

## 拉取/合并请求描述：(PR description)

[
ch569w-evt: add usbhs device mode driver

* usbd driver tested with cdc_vcom, internal loopback
  (can't run both MSH & usbd due to 16KB RAM limitation)
* reduce usrstack & main thread stack size for usbd test
* ch56x_uart.c : iron out UART0_PIN_ALT assignment
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

**Review by mysterywolf — APPROVED** [review]

File: bsp/wch/risc-v/ch569w-evt/applications/SConscript [hunk 1]
```diff
@@ -2,9 +2,11 @@ from building import *
 
 cwd = GetCurrentDir()
 
+app = ARGUMENTS.get('app', "main")
```
Comment (Guozhanxin) [inline bsp/wch/risc-v/ch569w-evt/applications/SConscript:5]: 这个为啥要改啊，啥意思啊

**Review by Guozhanxin — APPROVED** [review]

File: bsp/wch/risc-v/ch569w-evt/board/linker_scripts/link.lds [hunk 2]
```diff
@@ -180,8 +180,10 @@ SECTIONS
 	
 	.dmadata :
     {
-        . = ALIGN(16);
         PROVIDE( _dmadata_start = .);
+    	/* first 8 bytes are reserved for USB ep0 SETUP */
+		. = . + 8;
```
Comment (Guozhanxin) [inline bsp/wch/risc-v/ch569w-evt/board/linker_scripts/link.lds:185]: 这里的tab键要换成空格，要不然对不齐

File: bsp/wch/risc-v/ch569w-evt/applications/SConscript (hunk 1 above)
Comment (emuzit) [reply bsp/wch/risc-v/ch569w-evt/applications/SConscript:5]: 上次PR时mysterywolf要求main.c只保留LED闪烁，其他的测试放到test.c，这个修改是方便测试test.c：
    scons app=test

File: bsp/wch/risc-v/ch569w-evt/board/linker_scripts/link.lds (hunk 2 above)
Comment (emuzit) [reply bsp/wch/risc-v/ch569w-evt/board/linker_scripts/link.lds:185]: 这个文档是从WCH克隆过来修改的，没注意是用tab缩进。我把缩进改为空格好了。

**Pull request merged**